cpp/src/arrow/compute/exec/util.cc

```diff
@@ -383,5 +383,25 @@ size_t ThreadIndexer::Check(size_t thread_index) {
   return thread_index;
 }
 
+Status TableSinkNodeConsumer::Init(const std::shared_ptr<Schema>& schema,
+                                   BackpressureControl* backpressure_control) {
+  // If the user is collecting into a table then backpressure is meaningless
```

Comment:
Not related to this PR, but just a thought, should we move TableSourceNode to utils?

Reply:
The `SinkNodeConsumer` objects can be useful to add to the public API because users specify them as part of options.  That's why I had to move this one since the previous location wasn't public and I couldn't use it in tests.

The `TableSourceNode` is a node and those stay in the private API so I don't think we need to move that one.